src/modules/network/components/networkSwitcherDropdown/NetworkSwitcherDropdown.js

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable max-statements */
+import React, { useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import MenuSelect, { MenuItem } from '@wallet/components/MenuSelect';
+import Icon from 'src/theme/Icon';
+import useSettings from '@settings/hooks/useSettings';
+import { useBlockchainApplicationMeta } from '@blockchainApplication/manage/hooks/queries/useBlockchainApplicationMeta';
+import {
+  useApplicationManagement,
+  useCurrentApplication,
+} from '@blockchainApplication/manage/hooks';
+import { DEFAULT_NETWORK } from 'src/const/config';
+import { Client } from 'src/utils/api/client';
+import styles from './NetworkSwitcherDropdown.css';
+import networks from '../../configuration/networks';
+import { useNetworkStatus } from '../../hooks/queries';
+
+function NetworkSwitcherDropdown({
+  noLabel,
+  onLoadApplications = () => {},
+  onLoadNetworkStatus = () => {},
+}) {
+  const { t } = useTranslation();
+  const { setValue } = useSettings('mainChainNetwork');
+  const [selectedNetwork, setSelectedNetwork] = useState(networks[DEFAULT_NETWORK]);
+  const { setApplications } = useApplicationManagement();
+  const [, setCurrentApplication] = useCurrentApplication();
+  const queryClient = useRef(new Client({ http: selectedNetwork.serviceUrl }));
+
+  const networkStatus = useNetworkStatus({ client: queryClient.current });
+  const blockchainApps = useBlockchainApplicationMeta({
```

Comment:
`blockchainAppsMeta`
Better with full naming to not confuse since we also have a `blockchainApps` model in the BE. Also we have a bunch of `useBlockchainApplication...` hooks, so the last suffix feels relevant to keep.